contrib/windows/exe_path_wrapper.c

```diff
@@ -0,0 +1,73 @@
+#include <windows.h>
+#include <tchar.h>
+#include <stdio.h>
+#include <shlwapi.h>
+#define ENVVAR_MAXLEN 32760
```

Comment:
Actually, this is a lie (the documentation for windows environment variables is demonstrably wrong in this). There is no maximum environment variable length. Except for the PATH variable, which is generally limited to 1-2k before some things go haywire.

Reply:
Alright; I've opted to use the hardcoded limit you suggested below (`32768`) for all path elements and environment variables themselves.

Reply:
All environment variables except PATH have no limit. PATH is limited to 2k for maximum reliability.

Reply:
Okay looking into it more, it appears that if you set the environment variables using functions like `SetEnvironmentVariable (including `PATH`) it will set, but it will have issues when merging with the system PATH, especially if you launch `cmd.exe` as a sub-shell.

I will limit this to the de-factor limit of `1920`, as given in [this SO answer](https://superuser.com/questions/1070272/why-does-windows-have-a-limit-on-environment-variables-at-all).